lua/astrocommunity/pack/haskell/init.lua

```diff
@@ -12,33 +12,24 @@ return {
   },
   {
     "mrcjkb/haskell-tools.nvim",
-    ft = { "haskell" },
-    branch = "1.x.x", -- recommended by haskell-tools
-    init = function() astronvim.lsp.skip_setup = utils.list_insert_unique(astronvim.lsp.skip_setup, "hls") end,
-    opts = {
-      hls = {
-        on_attach = function(client, bufnr) require("astronvim.utils.lsp").on_attach(client, bufnr) end,
-      },
-    },
-    config = function(_, opts)
-      local tools = require "haskell-tools"
-      vim.api.nvim_create_autocmd("Filetype", {
-        pattern = "haskell", -- autocmd to start haskell-tools
-        callback = function() tools.start_or_attach(opts) end,
-      })
-
-      vim.api.nvim_create_autocmd("LspAttach", {
-        pattern = "*.hs", -- autocmd to start haskell-tools
-        callback = function(args)
-          local client = vim.lsp.get_client_by_id(args.data.client_id)
-          if client.name == "haskell-tools.nvim" then tools.dap.discover_configurations(args.buf) end
-        end,
-      })
-    end,
     dependencies = {
       "nvim-lua/plenary.nvim",
-      "nvim-telescope/telescope.nvim", -- optional
+      "nvim-telescope/telescope.nvim", -- Optional
     },
+    branch = "2.x.x", -- Recommended
+    init = function() -- Optional, see Advanced configuration
+      utils.list_insert_unique(astronvim.lsp.skip_setup, "hls")
+      vim.g.haskell_tools = {
```

Comment:
I just had a thought: I don't know if AstroNvim packages are designed to be user configurable.
It seems like they are, as `opts` can be a function that takes an `opts` parameter, which I'm assuming can be influenced by the user? (See for example the [java pack](https://github.com/AstroNvim/astrocommunity/blob/bb528993675119e4392acd1aea2b1691d5791005/lua/astrocommunity/pack/java/init.lua#L36)).

If this is the case, then I would suggest putting this line in the `init` function:

```lua
astronvim.lsp.skip_setup = utils.list_insert_unique(astronvim.lsp.skip_setup, "hls")
```

And setting `vim.g.haskell_tools` in the `config` function (merging the default with the `opts` passed to `config`).
I had recommended to set `vim.g.haskell_tools` in the `init` function, 
because lazy.nvim doesn't use Neovim's built-in initialisation order and I wanted to be on the safe side, setting `vim.g.haskell_tools` as early as possible). But iin principle it can also be set in the `config` function.

Note: As the previous package wasn't user-configurable, please don't see this as a blocker for this PR.

Reply:
It can, I don't think it should.

Reply:
@Uzaaft I presume you're referring to the fact that `vim.g.haskell_tools` can go in `config`?

What's your reasoning?

Reply:
All examples I've seen from folke does `vim.g` stuff in `init`

Reply:
That makes sense for plugins that have their Initialization in the `plugin` directory (see `:h runtimepath`), which is the majority of Vimscript plugins.

This is not the case for `haskell-tools`, which has its initialization in `ftplugin`.